### ISSUE TITLE
[bug] video artifacts happen when switching to monitors with transparent windows

### ISSUE THREAD

**Issue by bklaase** [herbstluftwm/herbstluftwm#1267]
Not sure, but I *think* it happened because of this fix: #1173 

video artifacts happen when switching to monitors with transparent windows. Or at perhaps 32 bit windows? happens with Zathura too for example. It's difficult to capture as a screenshot, but this happens:

![image](https://user-images.githubusercontent.com/8513117/113449893-1a2c8180-93ff-11eb-9e20-e6357a6de551.png)

for only a fraction of a second. It didn't happen in 9.2.0. 

reproduce by having zathura or termite, etc on a tag. switch to another and press and hold the shortcut to alternate between them. You'll spot it easilly. I checked and this is independent of compositor (even when not running any)

**Comment by bklaase** [comment]
update: on another system where I just updated to the latest git version, I don't have this issue. Both are on Intel graphics. What system information could help you?

**Comment by t-wissmann** [comment]
It doesn't happen on my nvidia setup, but it happens from time to time on my intel graphics laptop (but only on the decorations; the window content is never affected)

**Comment by Barbarossa93** [comment]
With the current release of picom, the decorations will stay for a second longer than the contents. In the current git, they go away immediately on window close. So this could possibly be a picom issue?

**Comment by bklaase** [comment]
That's what I tried as well, but the glitches show regardless of whether a compositor is active. It's very strange that the nature of the the artifacts is so different between systems. I have tried on 4 laptops, all with intel graphics. One has no issue at all. one is the screenshot above, one shows different glitches (more like rainbow noise) and one has glitching decorations, as Thorsten described.

All are on DRI2, with DRI driver: i965, and VDPAU driver: va_gl. I tested with other wm's too (BSPWM, awesome, openbox) and it doesn't happen theren. Also doesn't happen with the tagged hlwm release. Perhaps I should check out source from that tag onwards, until the issue emerges... :(

**Comment by t-wissmann** [comment]
Sorry that I have not looked into it yet; I'm very busy at the moment :-(
(Maybe this bug is introduced by https://github.com/herbstluftwm/herbstluftwm/pull/1236)

**Comment by bklaase** [comment]
not at all! you owe us nothing! My feigned unhappiness was purely directed at the prospect of checking out lots of commits, building and testing. But I will... :) 
Thanks for the pointer!

**Comment by tiosgz** [comment]
To add my observations… (probably nothing important)

I’ve been getting some decoration flickering recently, almost only when switching tags; less often when focusing another window. Changing the focus usually fixes it for me (until it reoccurs later).

Today also a whole window started doing the same, but only one. (I now have some hours of tag1=`(hsplit 0.6 luakit (vim in nnn in alacritty))`, tag2=`(max vivaldi-snapshot vivaldi-snapshot)` & Alacritty is the window that flickers. I guess this additional info is useless though.)

I’m running hlwm `0baa0318` right now with `picom-ibhagwan-git 1577_Next.296.g60eb00c_2021.02.08-1`; my laptop has an Intel integrated graphics card.
